projects/plugins/jetpack/extensions/shared/components/product-management-controls/use-products.js

```diff
@@ -0,0 +1,37 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { jetpackMembershipProductsStore } from './store';
+
+export default function useProducts( {
+	selectedProductId = 0,
+	setSelectedProductId = () => {},
+	shouldSkipResolver = false,
+} ) {
+	const products = useSelect( select => {
+		const { getProducts, getProductsNoResolver } = select( jetpackMembershipProductsStore );
+		return {
+			products: shouldSkipResolver
+				? getProductsNoResolver()
+				: getProducts( selectedProductId, setSelectedProductId ),
+		};
+	}, [] );
+
+	const { apiState, connectUrl, shouldUpgrade } = useSelect( select => {
+		const { getApiState, getConnectUrl, getShouldUpgrade } = select(
+			jetpackMembershipProductsStore
+		);
+		return {
+			apiState: getApiState(),
+			connectUrl: getConnectUrl(),
+			shouldUpgrade: getShouldUpgrade(),
+		};
+	} );
+
+	return { apiState, connectUrl, products, shouldUpgrade };
+}
```

Comment:
With the latest Redux refactor, this hook has basically become a simple helper if consumers need some of the data provided by the product management component.

All of this can be done directly with Redux if needed, but I think this is a slightly nicer, more controlled way of exposing a whole lot of complexity.

In my opinion, blocks shouldn't need to use this, but simply rely on the component itself.
Though, Premium Content is _complicated_ and has some logic relying on `apiState`, so I think this hook comes in handy for edge cases.